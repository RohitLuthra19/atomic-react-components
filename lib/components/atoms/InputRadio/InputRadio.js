// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import InputControl from '../InputControl';
import styles from './InputRadio.style';
import Label from '../Label';

type Props = {
  id ?: string,
  name: string,
  className ?: string,
  showLabelBeforeInput ?: boolean,
  labelProps ?: {},
  label ?: Node,
  children ?: Node
};

const InputRadio = ({
  id,
  className,
  showLabelBeforeInput,
  name,
  children,
  label,
  labelProps,
  ...otherProps
}: Props): Node => (
  <Label className={className} htmlFor={id} {...labelProps}>
    <InputControl id={id} type="radio" name={name} {...otherProps} />
    <span>{children || label}</span>
  </Label>
);

InputRadio.defaultProps = {
  id: '',
  className: '',
  labelProps: {},
  children: '',
  label: '',
  showLabelBeforeInput: false,
};

export default styled(InputRadio)`${styles}`;
export { InputRadio as InputRadioVanilla };
