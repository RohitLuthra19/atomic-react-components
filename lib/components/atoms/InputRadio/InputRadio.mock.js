const inputRadio = {
  name: 'test_1',
  id: 'test_1',
  value: 'yes',
  label: 'Default InputRadio',
};

const defaultCheckedInputRadio = {
  name: 'test_1',
  id: 'test_2',
  value: 'defaultChecked',
  checked: true,
  label: 'default checked',
};

const disabledInputRadio = {
  name: 'test_1',
  id: 'test_3',
  value: 'disabled',
  label: 'disabled',
  disabled: true,
};

const inputRadioWithLabelBeforeInput = {
  name: 'test_1',
  id: 'test_4',
  value: true,
  label: 'InputRadio with label before input',
  showLabelBeforeInput: true,

};

export {
  inputRadio,
  defaultCheckedInputRadio,
  disabledInputRadio,
  inputRadioWithLabelBeforeInput,
};
