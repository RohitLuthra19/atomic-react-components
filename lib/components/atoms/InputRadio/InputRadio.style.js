import { css } from 'styled-components';

export const inputStyles = css`
  position: absolute;
  opacity: 0;
  &:checked + label > span {
    &:after {
      background-color: ${props => props.theme.brandPrimaryColor};
    }
  }
  ${props => (props.disabled ? css`
    & + label > span {
      border-color: ${props.theme.fieldBorderDisabledColor};
    }
    & + label {
      cursor: default;
    }
  ` : css`
      &:focus + label > span,
      &:hover + label > span {
        border-color: ${props.theme.brandPrimaryColor};
      }
  `)}
`;

export const labelStyles = css`
  cursor: pointer;
  position: relative;
  display: inline-flex;
  align-items: center;
  > span {
    border: 2px solid ${props => props.theme.colorOutline};
    border-radius: 50%;
    width: 24px;
    height: 24px;
    position: relative;
    margin-right: 5px;

    &::after {
      content: "";
      width: 14px;
      height: 14px;
      border-radius: 50%;
      left: 50%;
      top: 50%;
      transform: translate(-50%,-50%);
      position: absolute;
    }
  }
`;
