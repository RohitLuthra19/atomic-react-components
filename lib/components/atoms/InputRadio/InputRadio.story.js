// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import InputRadio from './InputRadio';

storiesOf('Atoms', module).addWithChapters('InputRadio', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <div>
              <InputRadio name="test" id="test_1" value="yes">Yes</InputRadio>
              <InputRadio name="test" id="test_2" value="no">No</InputRadio>
              <InputRadio name="test" id="test_3" value="NA" disabled>Disabled</InputRadio>
            </div>
          ),
        },
        {
          sectionFn: () => (
            <div>
              <InputRadio id="test_html_1" name="test1" value="yes" checked showLabelBeforeInput><b>Yes</b></InputRadio>
              <InputRadio id="test_html_2" name="test1" value="no" showLabelBeforeInput><b>No</b></InputRadio>
            </div>
          ),
        },
      ],
    },
  ],
});
