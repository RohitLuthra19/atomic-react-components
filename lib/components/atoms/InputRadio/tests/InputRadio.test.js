import React from 'react';
import { shallow } from 'enzyme';
import { InputRadioVanilla } from '../InputRadio';

describe('InputRadio Component', () => {
  let inputRadio = '';
  beforeEach(() => {
    inputRadio = shallow(<InputRadioVanilla name="test" id="name">inputRadio</InputRadioVanilla>);
  });

  test('should render correctly', () => {
    expect(inputRadio).toMatchSnapshot();
  });

  test('should render disabled', () => {
    inputRadio = shallow(<InputRadioVanilla name="test" id="name" disabled>InputRadio</InputRadioVanilla>);
    expect(inputRadio.find({ disabled: true })).toHaveLength(1);
  });

  test('should render checked', () => {
    inputRadio = shallow(<InputRadioVanilla name="test" id="name" checked>InputRadio</InputRadioVanilla>);
    expect(inputRadio.find({ checked: true })).toHaveLength(1);
  });

  test('should render label as children if label prop is passed', () => {
    inputRadio = shallow(<InputRadioVanilla name="test" id="name" label="inputRadio label" />);
    expect(inputRadio.find('span').text()).toEqual('inputRadio label');
  });
});
