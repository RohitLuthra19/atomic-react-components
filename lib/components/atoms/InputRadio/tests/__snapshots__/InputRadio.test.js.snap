// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadio Component should render correctly 1`] = `
<Label
  className=""
  htmlFor="name"
>
  <InputControl
    id="name"
    name="test"
    type="radio"
  />
  <span>
    inputRadio
  </span>
</Label>
`;
